lazyflow/distributed/TaskOrchestrator.py

```diff
@@ -0,0 +1,78 @@
+from mpi4py import MPI
+from threading import Thread
+import enum
+from typing import Generator, TypeVar, Generic, Callable, Tuple
+import uuid
+
+END = "END-eb23ae13-709e-4ac3-931d-99ab059ef0c2"
+TASK_DATUM = TypeVar("TASK_DATUM")
+
+
+@enum.unique
+class Tags(enum.IntEnum):
+    TASK_DONE = 13
+    TO_WORKER = enum.auto()
+
+
+class _Worker(Generic[TASK_DATUM]):
+    def __init__(self, comm, rank: int):
+        self.comm = comm
+        self.rank = rank
+        self.stopped = False
+
+    def send(self, datum: TASK_DATUM):
+        print(f"Sending datum {datum} to worker {self.rank}...")
+        self.comm.send(datum, dest=self.rank, tag=Tags.TO_WORKER)
+
+    def stop(self):
+        self.send(END)
+        self.stopped = True
+
+
+class TaskOrchestrator(Generic[TASK_DATUM]):
+    def __init__(self, comm=None):
+        self.comm = comm or MPI.COMM_WORLD
+        self.rank = self.comm.Get_rank()
+        num_workers = self.comm.size - 1
+        if num_workers <= 0:
+            raise Exception("Trying to orchestrate tasks with no workers!!!")
+        self.workers = {rank: _Worker(self.comm, rank) for rank in range(1, num_workers + 1)}
+
+    def get_finished_worker(self) -> _Worker[TASK_DATUM]:
+        status = MPI.Status()
+        self.comm.recv(source=MPI.ANY_SOURCE, tag=Tags.TASK_DONE, status=status)
+        return self.workers[status.Get_source()]
+
+    def orchestrate(self, task_data: Generator[TASK_DATUM, None, None]):
+        print(f"ORCHESTRATOR: Starting orchestration of {len(self.workers)}...")
```

Comment:
do you rely on printing? or was that just for debugging? might consider switching to logging.